macros/cargo

```diff
@@ -16,9 +16,10 @@
 %__cargo_cross_opts %{__cargo_common_opts} --target %{__cargo_target}
 %__cargo_cross_opts_static %{__cargo_common_opts} --target %{__cargo_target_static}
 %__cargo_env CARGO_TARGET_DIR="${HOME}/.cache" SKIP_README="true"
+%__cargo_env_static CARGO_TARGET_DIR="${HOME}/.cache/static" SKIP_README="true"
```

Comment:
I'm slightly worried about the possibility of cargo using a directory named "static" under its target dir someday, though it's safe for now.  ".cache_static" to avoid any worry?

Similarly, we know skipping readme builds here is safe because we're only calling it for two things after our normal builds are done, but that may not always be true.  How about at least adding a comment here, like "# Take advantage of the fact that cargo_build_static is only used after cargo_build, so we don't need to rebuild READMEs" ?

Reply:
We skip READMEs because the build scripts will attempt to write them to the same directory as `build.rs` and `Cargo.toml`, which is immutable at this point.

We have to use a subdirectory of `.cache` or else we'd need a second cache layer to wire up, which feels excessive. Maybe `.cache/.static` instead?

Reply:
> We skip READMEs because the build scripts will attempt to write them to the same directory as build.rs and Cargo.toml, which is immutable at this point.

Sorry, I was silly and blind, it's also on the line above. :)

> We have to use a subdirectory of .cache or else we'd need a second cache layer to wire up, which feels excessive. Maybe .cache/.static instead?

Ah, right, I forgot about the cache layer being mounted there.  I don't think it's a big risk, but we could be more specific if you want.  `.static`, or `.buildsys_static` to be fully safe perhaps.  I'm also totally fine with it as-is.